File: SDKs/amazonaws.com/organizations/2016-11-28/python/src/sdk/models/shared/child.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class Child:
    r"""Child
    Contains a list of child entities, either OUs or accounts.
    """
    
    id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Id') }})
    type: Optional[ChildTypeEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Type') }})
    

File: SDKs/amazonaws.com/organizations/2016-11-28/typescript/src/sdk/models/shared/child.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ChildTypeEnum } from "./childtypeenum";



// Child
/** 
 * Contains a list of child entities, either OUs or accounts.
**/
export class Child extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=Type" })
  type?: ChildTypeEnum;
}
